src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AttributeCollection.cs

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Collections;
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA2112:SecuredTypesShouldNotExposeFields", Scope = "type", Target = "System.ComponentModel.AttributeCollection")]
+
+namespace System.ComponentModel
+{
+    /// <devdoc>
+    ///     Represents a collection of attributes.
+    /// </devdoc>
+    public class AttributeCollection : ICollection
+    {
+        /// <devdoc>
+        ///     An empty AttributeCollection that can used instead of creating a new one.
+        /// </devdoc>
+        public static readonly AttributeCollection Empty = new AttributeCollection(null);
+        private static Hashtable s_defaultAttributes;
+
+        private Attribute[] _attributes;
+
+        private static readonly object s_internalSyncObject = new object();
+
+        private struct AttributeEntry
+        {
+            public Type type;
+            public int index;
+        }
+
+        private const int FOUND_TYPES_LIMIT = 5;
+
+        private AttributeEntry[] _foundAttributeTypes;
+
+        private int _index = 0;
+
+        /// <devdoc>
+        ///     Creates a new AttributeCollection.
+        /// </devdoc>
+        public AttributeCollection(params Attribute[] attributes)
+        {
+            _attributes = attributes ?? new Attribute[0];
+
+            for (int idx = 0; idx < _attributes.Length; idx++)
+            {
+                if (_attributes[idx] == null)
+                {
+                    throw new ArgumentNullException(nameof(attributes));
+                }
+            }
+        }
+
+        protected AttributeCollection()
+        {
+        }
+
+        /// <devdoc>
+        ///     Creates a new AttributeCollection from an existing AttributeCollection
+        /// </devdoc>
+        public static AttributeCollection FromExisting(AttributeCollection existing, params Attribute[] newAttributes)
+        {
+            // VSWhidbey #75418
+            // This method should be a constructor, but making it one introduces a breaking change.
+            // 
+            if (existing == null)
+            {
+                throw new ArgumentNullException(nameof(existing));
+            }
+
+            if (newAttributes == null)
+            {
+                newAttributes = new Attribute[0];
```

Comment:
I had to revert the change - it's only available on .NET 4.6+ and fails to compile on .NET Standard 1.0.